crmsh/ui_resource.py

```diff
@@ -497,11 +498,54 @@ def do_constraints(self, context, resource):
 
     @command.wait
     @command.completers(compl.resources, _attrcmds, compl.nodes)
-    def do_failcount(self, context, rsc, cmd, node, value=None):
+    def do_failcount(self, context, rsc, cmd, node, value=None, operation=None, interval=None):
         """usage:
-        failcount <rsc> set <node> <value>
+        failcount <rsc> set <node> <value> [operation] [interval]
         failcount <rsc> delete <node>
         failcount <rsc> show <node>"""
+        if cmd == "set":
+            nodeid = utils.get_nodeid_from_name(node)
+            if nodeid is None:
+                context.fatal_error("Node {} not in this cluster".format(node))
+
+            # query the current failcount status
+            query_cmd = "cibadmin -Ql --xpath '/cib/status/node_state[@id='{}']'".format(nodeid)
+            rc, out, err = utils.get_stdout_stderr(query_cmd)
+            if rc != 0:
+                context.fatal_error(err)
+
+            # try to get failcount dict {operation:interval}
+            import re
+            failcount_res = re.findall(r'fail-count-{}#(.*)_([0-9]+)'.format(rsc), out)
+            if not failcount_res:
+                context.fatal_error("No failcount on node {}".format(node))
+            failcount_dict = dict(failcount_res)
+
+            # validate for operation and interval
+            if operation and operation not in failcount_dict.keys():
+                context.fatal_error("Should specify operation between \"{}\"".format(' '.join(failcount_dict.keys())))
+            if (operation and interval) and (operation, interval) not in failcount_res:
+                context.fatal_error("Should specify (operation, interval) between {}".format(failcount_res))
```

Comment:
We don't need to prevent users from setting a fail-count that is not existing yet.

Reply:
But setting a non-exist entry make no sense.
And validate here also will be helpful for misspelling.